invoices/invoiceregistry.go

```diff
@@ -652,12 +652,6 @@ func (i *InvoiceRegistry) processKeySend(ctx invoiceUpdateCtx) error {
 		return errors.New("invalid keysend preimage")
 	}
 
-	// Don't accept zero preimages as those have a special meaning in our
```

Comment:
May be nice to leave this check in place, so we can catch this earlier. IIUC, with this new set of changes, we'll catch this when we go to _serialize_ the new invoice.

Reply:
Yes it is caught in the database now. Why is it nice to check twice?

Reply:
Meant more so to possibly lave it at this level if we think there's an advantage in rejecting it at this level before having to open a database transaction. It's a question of how much "biz logic" we want to have in the database vs ensuring that things are well formed _before_ inserting them into the database.

Reply:
In general, I think it is safest to check at the db level to be absolutely sure we never corrupt the db by higher level code that 'forgets' to do this check.